### ISSUE TITLE
after change frame_acq_time to 1sec,   xrun(0, 5),     60sec scan finished in 6 sec.

### ISSUE THREAD

**Issue by adelezh** [xpdAcq/mission-control#118]
![img_7373](https://user-images.githubusercontent.com/32132871/41053466-97692b38-6989-11e8-99c5-eae39b4711b7.jpg)

**Comment by sbillinge** [comment]
Thanks for reporting, we will look into it.

@chiahaoliu it looks as if the INFO statement after the frame rate is
changed is also incorrect.  It talks about "exposure"

**Comment by chiahaoliu** [comment]
yeah I've contacted the beamline and we are trying to figure out.

I will update print statement in acq too.

**Comment by chiahaoliu** [comment]
More details:

This error was from new detector ``pe3``. Since a 20-hr experiment is still running at this moment, we couldn't get a chance to fully test the issue.

**Comment by chiahaoliu** [comment]
After a more in-depth test with @jrmlhermitte and @sghose this evening, we were not able to reproduce the same issue reported here but we did find a glitch at control side (PV) which is likely be the origin of this problem.

* Short story about the result:

  After the test (described below), we were not able to reproduce the issue. But @jrmlhermitte found a glitch at the control level. He noticed once detector is staged or still in continuous acquisition mode, even though it's still possible to configure `acquire_time` and have correct readback with `get`  method at `ophyd` level, the configuration was in fact ignored by the hardware and machine status remained unchanged.  This could very likely be the root of this issue.

  When `glbl['frame_acq_time'] = 1` was called, xpdAcq stops continuous acquisition mode, update new frame acquisition time, wait for 1s to let the configuration kick in, and restarts continuous acquisition mode. The 1s wait time was determined empirically ~2 yrs ago. When the network isn't stable, it's possible the internal wait time isn't sufficient for propagating this signal to IOC server, detector and back. Once that happen, the glitch described above would cause the issue reported: `ophyd` reports 1s acquire time therefore a 60s scan means summing 60 frames but the machine was in fact in the previous state, 0.1s per frame, and summing 60 frames only gave a 6s scan.

There are two things we maybe able do for this issue:
1. Control side definitely needs to be fixed. I believe request was sent by beamline staffs.
2. Increase the wait time when setting frame acquisition time in xpdAcq.

Using more sophisticated approach like `set` or `configure` method instead of `put` method in current backend might also help too (I am not familiar with this)

attn: @dooryhee @adelezh 

* Long story about the test

  The total exposure time of PE detectors at XPD are configured by two attributes:
``pe1c.images_per_set`` (how many frames to be summed into a image) and ``pe1c.cam.acquire_time``(how long is a frame). We can configure `images_per_set` while detector is acquiring but we need to stop the continuous acquisition mode when setting `acquire_time` (and turn it back). The product of these two values should match total exposure time. This issue reported here is about incorrect total exposure time. Our test to this problem can be described as following.

  We first tried configuring ``pe1c.images_per_set``  at ``ophyd`` level and make sure the IOC and `ophyd` object readback values from the device were correct. We ran a bluesky `count` scan with default `RE` and `xrun`, detector worked fine in both cases. 

  xpdAcq syntax `glbl['frame_acq_time'] = 1` was then used to configure acquire time per frame (as this issue) and again confirmed the IOC and `ophyd` object readback values from the device were correct. A `xrun` with xpdAcq customized ScanPlan and Sample objects was run, this scan ran properly.

  A series of  `glbl['frame_acq_time']` was set and there was always a xrun call right after changing acquire time for testing. No issue was observed. Up to this point, we were not able to reproduce the issue.

**Comment by adelezh** [comment]
pe3 works normal after we change 'frame_acq_time' from 0.1 sec to 1 sec, and from 1sec to 0.1 sec.

**Comment by adelezh** [comment]
pe2 works normal after we change 'frame_acq_time' from 0.1 sec to 1 sec, and from 1sec to 0.1 sec.

**Comment by CJ-Wright** [comment]
Hmm that is very interesting. Maybe it is a detector level issue?

**Comment by sbillinge** [comment]
Did anyone ever reproduce this error?
